README.md

```diff
@@ -1,2 +1,285 @@
+👑 Queenbee is a workflow language for creating complex workflows! The Queenbee workflow
+is inspired by [Argo Workflow](https://argoproj.github.io/docs/argo/readme.html) and
+borrows a number of terms and expressions from
+[Apache Airflow](http://airflow.apache.org/) and [Ansible](https://docs.ansible.com/).
+
+Queenbee populates and validates the workflows but does not run them. For running the
+workflows see `ladybug-tools/workerbee` which converts Queenbee workflows to executable
+[Luigi](https://luigi.readthedocs.io/en/stable/) workflows.
+
+You can find more workflow samples in
+[honeybee-radiance-workflow](https://github.com/ladybug-tools/honeybee-radiance-workflow)
+repository. The workflow is meant to mainly be used with Ladybug Tools libraries but it
+should also support any other custom workflows.
+
+# Queenbee Workflow structure
+
+A Queenbee workflow is a YAML / JSON file which is consisted from different parts. You
+can use Queenbee to generate the workflows programmatically using Python or you can
+write a workflow line by line.
+
+Here we discuss the generic structure of a Queenbee workflow. See OpenApi schema
+documentation for all Workflow objects: [Missing link]()
+
+```
+
+  Workflow
+    |
+    |__ inputs
+    |   |___ parameters
+    |   |___ artifacts
+    |
+    |__ operators
+    |
+    |__ templates
+    |
+    |__ flow
+    |   |___ tasks / steps
+    |
+    |__ outputs
+        |___ parameters
+        |___ artifacts
+
+```
+
+## 1. inputs
+
+Workflow inputs are global inputs that can be accessed using `"{{workflow.inputs.xx.yy}}"`.
+For instance to access an input parameter with the name `file_path` one should use
+`"{{workflow.inputs.parameters.file_path}}"` and Queenbee will map the value of
+`file_path` parameter to this placeholder.
+
+Inputs can be from two different types: `parameters` and `artifacts`. Use `artifacts` for
+*files* and *folder* inputs. The content of these `artifacts` will be copied to target
+location before executing the workflow. For all the other `inputs` use `parameters`.
+
+Here is an example `inputs` for a daylight simulation workflow. You can put a default
+value for inputs or it can be passed to workflow using an input YAML / JSON file.
+
+```yaml
+
+  inputs:
+    parameters:
+    # this is just the path to target folder not the content itself. It can be referenced
+    # as {{workflow.parameters.folder}}
+    - name: folder
+      value: /mnt/c/ladybug/untitled
+    artifacts:
+    - name: sky_file
+      value: full_path_to_sky_file.sky
+    - name: material_file
+      value: full_path_to_material_file.mat
+    - name: geometry_file
+      value: full_path_to_geometry_file.rad
+
+```
+
+## 2. operators
+
+`Operators` include the requirements for running `templates` locally or using a Docker
+image. This is an operator for running arbitrary Radiance commands.
+
+```yaml
+
+operators:
+  - name: radiance-operator
+    inputs:
+      parameters:
+        - name: command
+    container:
```

Comment:
Is this a mandatory field?

Reply:
no. only name is at this point which means you can create an operator that only runs locally or only runs using a container. Maybe I should break down operators to be two different types for `LocalOperator` and `DockerOperator`?

Reply:
Nah i think it's fine to keep them in as optional rather than two different types of operators.

Reply:
Thanks. This is how it ended up being. see the updated documentation.